kedro/extras/datasets/datatable/csv_dataset.py

```diff
@@ -0,0 +1,178 @@
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
+            credentials: Credentials required to get access to the underlying filesystem.
+                E.g. for ``GCSFileSystem`` it should look like `{"token": None}`.
+            fs_args: Extra arguments to pass into underlying filesystem class constructor
+                (e.g. `{"project": "my-project"}` for ``GCSFileSystem``), as well as
+                to pass to the filesystem's `open` method through nested keys
+                `open_args_load` and `open_args_save`.
+                Here you can find all available arguments for `open`:
+                https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.spec.AbstractFileSystem.open
+        """
+        _fs_args = copy.deepcopy(fs_args) or {}
+        _fs_open_args_load = _fs_args.pop("open_args_load", {})
+        _fs_open_args_save = _fs_args.pop("open_args_save", {})
+        _credentials = copy.deepcopy(credentials) or {}
+
+        protocol, path = get_protocol_and_path(filepath, version)
+
+        self._protocol = protocol
+        self._fs = fsspec.filesystem(self._protocol, **_credentials, **_fs_args)
+
+        super().__init__(
+            filepath=PurePosixPath(path),
+            version=version,
+            exists_function=self._fs.exists,
+            glob_function=self._fs.glob,
+        )
+
+        # Handle default load and save arguments
+        self._load_args = copy.deepcopy(self.DEFAULT_LOAD_ARGS)
+        if load_args is not None:
+            self._load_args.update(load_args)
+        self._save_args = copy.deepcopy(self.DEFAULT_SAVE_ARGS)
+        if save_args is not None:
+            self._save_args.update(save_args)
+
+        _fs_open_args_load.setdefault("mode", "r")
+        _fs_open_args_save.setdefault("mode", "w")
+        self._fs_open_args_load = _fs_open_args_load
+        self._fs_open_args_save = _fs_open_args_save
+
+    def _describe(self) -> Dict[str, Any]:
+        return dict(
+            filepath=self._filepath,
+            protocol=self._protocol,
+            load_args=self._load_args,
+            save_args=self._save_args,
+            version=self._version,
+        )
+
+    def _load(self) -> dt.Frame:
+        load_path = get_filepath_str(self._get_load_path(), self._protocol)
+
+        with self._fs.open(load_path, **self._fs_open_args_load) as fs_file:
+            return dt.fread(file=fs_file, **self._load_args)
+
+    def _save(self, data: dt.Frame) -> None:
+        save_path = get_filepath_str(self._get_save_path(), self._protocol)
+
+        with self._fs.open(save_path, **self._fs_open_args_save) as fs_file:
+            # convert to pandas before saving as otherwise can not use fs_file
+            data.to_pandas().to_csv(path_or_buf=fs_file, **self._save_args)
```

Comment:
Why are you still using `pandas` to save the data?

Reply:
Indeed, that is inefficient and wierd.
The reason is that datatable doesn't seem to support file-like object as input for writing out a table. Therefore one can not benefit from access to multiple various filesystems that is provided by `fsspec.open()`

Reply:
Right, so the main purpose of this dataset would be that you can load data in `datatable` and then work with it, before saving it again as pandas? If so, I'd suggest to update the description of the class to make this clear.

Reply:
yes, correct. the main 2 benefits would be:
* speed-up of csv reading, as `datatable` allows multi-thread reading from csv
* usage of data manipulation API that is familiar to R users, who might not have experience with `pandas`.
There is a nice list of advantages summarised in the original issue (#592).

Ok, i'll add doc-string of the class. This and the other proposed change will take a couple of days due to other tasks on my TODO.

Reply:
ok, the changed have been pushed

Reply:
According to [`Frame.to_csv()`](https://datatable.readthedocs.io/en/latest/api/frame/to_csv.html) docs, "If no path is given, then the Frame will be serialized into a string, and that string will be returned".

So maybe we should just get such string, `.encode("utf-8")` it and send to fsspec instead of converting to pandas?

Reply:
Unfortunately, I do not have enough experience to judge on it. I can blindly implement your suggestion, but I will not be able to validate it due to lack of time.

Reply:
Unfortunately, I do not have enough experience to judge on it. I can blindly implement your suggestion, but I will not be able to validate it due to lack of time.